tests/integration/api_packages_generic_test.go

```diff
@@ -98,6 +104,37 @@ func TestPackageGeneric(t *testing.T) {
 		})
 	})
 
+	t.Run("List", func(t *testing.T) {
+		defer tests.PrintCurrentTest(t)()
+
+		req := NewRequest(t, "GET", listURL)
+		resp := MakeRequest(t, req, http.StatusOK)
+
+		var expected []generic.PackageInfo
+		err := json.Unmarshal(resp.Body.Bytes(), &expected)
+		assert.NoError(t, err)
+
+		assert.Len(t, expected, 1)
+		assert.Len(t, expected[0].Files, 2)
+
+		resPkg := expected[0]
+		assert.Equal(t, packageVersion, resPkg.Version)
+		assert.Equal(t, int64(0), resPkg.DownloadCount)
+
+		// json results are ordered differently in different db engines for some reason
```

Comment:
This should be better fixed in the  GetFilesByVersionID model function (might be it's missing OrderBy?)

Reply:
The function itself does not have an orderby parameter nor an `OrderBy` operation in the db query in its [current implementation](https://github.com/go-gitea/gitea/blob/23a37b4b77ba7ec7f1079fe5b573fe4cc10fb142/models/packages/package_file.go#L69-L72):

```golang
// GetFilesByVersionID gets all files of a version
func GetFilesByVersionID(ctx context.Context, versionID int64) ([]*PackageFile, error) {
	pfs := make([]*PackageFile, 0, 10)
	return pfs, db.GetEngine(ctx).Where("version_id = ?", versionID).Find(&pfs)
}
```

I'm not sure if whether the problem had to do with the IDs or not, though its certainly possible.

Reply:
I can't seem to test for this myself. The mysql unit-tests fail to run at all on my host machine for some reason, even after explicitly following [the directions provided](https://github.com/go-gitea/gitea/blob/main/tests/integration/README.md#run-mysql-integration-tests):

<img width="980" height="311" alt="image" src="https://github.com/user-attachments/assets/321b5abe-8dc4-422c-9248-2d9bf4ea3fee" />

Reply:
alright, because I had the time whilst waiting for additional reviews and feedback, after a lot of substantial testing locally  it does not appear that changing the db function effects any of the unit-tests so far as I can tell. That seems to suggest that everything should be just fine and dandy if we do make that change. I was worried it was going to break a lot of things.

so I've decided to just go for it and make the change to put the sorting in the db function.